packages/b/box2d/xmake.lua

```diff
@@ -0,0 +1,113 @@
+package("box2d")
+
+    set_homepage("https://box2d.org")
+    set_description("A 2D Physics Engine for Games")
+
+    if is_plat("windows", "linux", "macosx", "mingw") then
+        set_urls("https://github.com/erincatto/box2d/archive/v$(version).zip")
+        add_versions("2.4.0", "6aebbc54c93e367c97e382a57ba12546731dcde51526964c2ab97dec2050f8b9")
+    end
+
+    add_deps("cmake")
+
+    on_load(function (package)
+        if is_plat("windows") and package:config("build_testbed") then
+            package:add("syslinks", "User32", "Gdi32", "Shell32")
+        end
+    end)
+
+    add_configs("build_unit_tests", {description = "Build the Box2D unit tests", default = true, type = "boolean"})
+    add_configs("build_testbed",    {description = "Build the Box2D testbed", default = false, type = "boolean"})
+    add_configs("build_docs",       {description = "Build the Box2D documentation", default = false, type = "boolean"})
+
+    on_install("windows", "linux", "macosx", "mingw", function (package)
+        local config = {}
+        local function add_config_arg(config_name, cmake_name)
+            table.insert(config, "-D" .. cmake_name .. "=" .. (package:config(config_name) and "ON" or "OFF"))
+        end
+
+        add_config_arg("build_unit_tests", "BOX2D_BUILD_UNIT_TESTS")
+        add_config_arg("build_testbed",    "BOX2D_BUILD_TESTBED")
+        add_config_arg("build_docs",       "BOX2D_BUILD_DOCS")
+
+        if xmake.version():ge("2.3.7") then
+            import("package.tools.cmake").build(package, config, {buildir = "build"})
+        else
+            table.insert(config, "-S")
+            table.insert(config, ".")
+            table.insert(config, "-B")
+            table.insert(config, "build")
+
+            os.mkdir("build")
+            os.vrunv("cmake", config)
+
+            if is_plat("windows") then
+                local old_dir = os.cd("build")
+
+                local configs = {}
+                local arch = package:is_arch("x86") and "Win32" or "x64"
+                local mode = package:debug() and "Debug" or "Release"
+
+                table.insert(configs, "/property:Configuration=" .. mode)
+                table.insert(configs, "/property:Platform=" .. arch)
+                table.insert(configs, "box2d.sln")
+
```

Comment:
we need set "/MT" or "/MD" for vs runtime.

```lua
        local vs_runtime = package:config("vs_runtime")
        if vs_runtime then
            table.insert(configs, '-DCMAKE_CXX_FLAGS_DEBUG="/' .. vs_runtime .. 'd"')
            table.insert(configs, '-DCMAKE_CXX_FLAGS_RELEASE="/' .. vs_runtime .. '"')
            table.insert(configs, '-DCMAKE_C_FLAGS_DEBUG="/' .. vs_runtime .. 'd"')
            table.insert(configs, '-DCMAKE_C_FLAGS_RELEASE="/' .. vs_runtime .. '"')
        end
```

see https://github.com/xmake-io/xmake/blob/e49ef767d42e4e4cfd1ecf659e5fd1ea4b8fa9f9/xmake/modules/package/tools/cmake.lua#L30-L37